extmod/nimble/hal/hal_uart.c

```diff
@@ -86,22 +91,36 @@ int hal_uart_close(uint32_t port) {
     return 0; // success
 }
 
+STATIC void mp_bluetooth_hci_uart_char_cb(uint8_t chr) {
+    #if HCI_TRACE
+    printf("> %02x\n", chr);
+    #endif
+    hal_uart_rx_cb(hal_uart_rx_arg, chr);
+}
+
 void mp_bluetooth_nimble_hci_uart_process(bool run_events) {
     bool host_wake = mp_bluetooth_hci_controller_woken();
 
-    int chr;
-    while ((chr = mp_bluetooth_hci_uart_readchar()) >= 0) {
-        #if HCI_TRACE
-        printf("> %02x\n", chr);
+    do {
```

Comment:
I think the `do {} while(1)` is equivalent to just `while(1) {}` here.

Reply:
Yeah true, I initially thought the `do` syntax is nice to show it always runs (at least once) but `while (1)` really is equally clear.

Reply:
The style in MicroPython is to use `for (;;) { ... }`